geps/gep-3614/index.md

```diff
@@ -0,0 +1,81 @@
+# GEP-3614: Firewall
+
+* Issue: [#3614](https://github.com/kubernetes-sigs/gateway-api/issues/3614)
+* Status: Provisional
+
+## TLDR
+
+The ability to attach firewall rules for ingress L3, L4 and L7 Gateway traffic.
+
+## Motivation
+
+`Gateways` are commonly exposed to the internet, which puts them as risk of
+attack. Internal networks can become compromised as well. We should provide
+tooling, documentation and best-practices for users to restrict and control
+access to their `Gateways`.
+
+### Definitions
+
+* "Firewall Engine" - A processor of request payloads and applies rulesets to
+  the contents to identify malicious, anomalous or otherwise unwanted traffic.
+  These are generally at the front of the request path, and may be attached to a
+  `Gateway` as a sidecar, integrated natively as part of the `Gateway`, or
+  deployed in front of the `Gateway` as part of the networking path.
+
+### User Stories
+
+* As an application developer, I want to allow specific IPs to access my
+  application.
+* As an application developer, I want to block or allow requests based on
+  headers; e.g. allow or deny specific user-agents.
+* As a gateway operator I want to be able to identify and block and log
+  malformed HTTP requests before they reach backend applications.
+* As a gateway operator I want to be able to provide my own signature-based
```

Comment:
* As a gateway operator I want to mutate (or block, or log) requests that may contain PII on responses (optional / very extended)

Reply:
Added it with some minor tweaks, and from the perspective of a "compliance officer", Might be a little too specific, but one thing it importantly highlights is that there will be needs for _response_ handling as well as requests. LMKWYT

Reply:
Not sure if it may be worth referencing response scope as part of DLP (data loss prevention)/egress use cases?